### PR TITLE
Pos eth refuel

### PR DESCRIPTION
**Pull request by nitinmittal23**

### CODE REVIEW

File: src/pos/erc20.ts [hunk 1]
```diff
@@ -151,6 +151,34 @@ export class ERC20 extends POSToken {
         });
     }
 
+    private depositEtherWithGas_(amount: TYPE_AMOUNT, userAddress: string, swapEthAmount: TYPE_AMOUNT, swapCallData: string, option: ITransactionOption = {}) {
+        this.checkForRoot("depositEtherWithGas");
+
+        return this.getChainId().then((chainId: number) => {
+            if (chainId !== 1) {
```
Comment (saranonearth) [inline src/pos/erc20.ts:158]: what if I am testing on goerli(5), should it tolerate testnet also?

Comment (nitinmittal23) [reply src/pos/erc20.ts:158]: its not possible on testnet.

Comment (nitinmittal23) [reply src/pos/erc20.ts:158]: feature is only for mainnet

**Review by saranonearth — APPROVED** [review]
lgtm

**Review by saranonearth — APPROVED** [review]
gucci

**Pull request merged**